### PR TITLE
[INF] simplify a bit linting, use pre-commit as CI linting checker

### PR DESCRIPTION
**Pull request by Zeroto521**
# PR Description

Please describe the changes proposed in the pull request:

There is two similar linting CIs (pre-commit and code-checks) but also have tiny differences.
We should use one of them as the standard, not both. Otherwise, we would get confused.

# PR Checklist

Please ensure that you have done the following:

1. [x] PR in from a fork off your branch. Do not PR from `<your_username>`:`dev`, but rather from `<your_username>`:`<feature-branch_name>`.

2. [x] If you're not on the contributors list, add yourself to `AUTHORS.rst`.

3. [x] Add a line to `CHANGELOG.md` under the latest version header (i.e. the one that is "on deck") describing the contribution.
    - Do use some discretion here; if there are multiple PRs that are related, keep them in a single line.

# Automatic checks

There will be automatic checks run on the PR. These include:

- Building a preview of the docs on Netlify
- Automatically linting the code
- Making sure the code is documented
- Making sure that all tests are passed
- Making sure that code coverage doesn't go down.

# Relevant Reviewers

Please tag maintainers to review.

- @ericmjl

### CODE REVIEW

File: .pre-commit-config.yaml
Comment (Zeroto521) [inline .pre-commit-config.yaml:30]: code-checks.yml has flake8, but pre-commit doesn't.

File: .pre-commit-config.yaml
Comment (Zeroto521) [inline .pre-commit-config.yaml]: code-checks.yml's config is complicated, but pre-commit's config is simple.

**Comment by Zeroto521** [comment]
Because the linting standard is different, so https://github.com/pyjanitor-devs/pyjanitor/pull/892/checks is red.

Now we should determine to use which linting is the standard.
code-checks.yml or pre-commit?

**Comment by ericmjl** [comment]
@Zeroto521 thanks for handling this! In the interest of minimizing the diff, I think we should defer to the codechecks.yml file for the settings, but use pre-commit as our actual code checker. Would you be open to making that happen please?

File: .pre-commit-config.yaml
Comment (Zeroto521) [inline .pre-commit-config.yaml:20]: nbstripout should be removed.
It changed a lot, even would make ipynb outputs become empty.

**Review by Zeroto521 — COMMENTED** [review]
Now use the old code-checks.yml as the base standard and pre-commit as the lining checker.

But these hooks weren't removed. They changed a little. Just remove some whitespace when lint.
```yaml
hooks:
  - id: trailing-whitespace
  - id: end-of-file-fixer
  - id: check-yaml
  - id: check-added-large-files
```

File: .pre-commit-config.yaml
Comment (ericmjl) [reply .pre-commit-config.yaml:20]: Got it, thanks @Zeroto521.

We can remove it for now. In a future setting, though, I think we'd want to do the notebooks such that they are built in the CI system and deployed.

**Comment by ericmjl** [comment]
Let's get one more review in before merging. Tagging @loganthomas and @hectormz, whom I think might take an interest in the PR.

**Review by ericmjl — APPROVED** [review]

File: .pre-commit-config.yaml
Comment (Zeroto521) [reply .pre-commit-config.yaml:20]: I see conf.py already adds nbsphinx, but no a `.rst` file index these notebooks.

Comment (ericmjl) [reply .pre-commit-config.yaml:20]: Right. Regarding the notebooks (and docs more generally), I have been debating, btw, to switch the docs over to mkdocs + mkdocs-material. It is much easier to maintain in the long-run than Sphinx; I know I have been hesitant to make any changes to the docs as it stands right now because of a lot of magic that happens behind the scenes that I'm just not well-versed in, whereas MkDocs and mkdocs-material are much easier to maintain combined. (It feels... flatter.)

**Comment by ericmjl** [comment]
@Zeroto521 I'm going to merge as I'd like to have these changes present in #897. Thanks for your contributions here, much appreciated!

**Pull request merged**
